api/v1alpha1/virtualbackend_types.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	// KindVirtualBackend is the name of the VirtualBackend kind.
+	KindVirtualBackend = "VirtualBackend"
+)
+
+// +kubebuilder:validation:Minimum=100
+// +kubebuilder:validation:Maximum=599
+type StatusCode uint32
+
+// +kubebuilder:validation:Pattern=`^([\w-]+)$`
+type ResponseHeader string
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:categories=envoy-gateway,shortName=vb
+//
+// VirtualBackend defines the configuration for direct response.
+type VirtualBackend struct {
```

Comment:
@guydc do you thinks it's a good idea to support this in `Backend`?

Reply:
Sorry for the very late response. IMHO, the best direction here is to support this through a custom filter, as outlined here: https://github.com/envoyproxy/gateway/issues/4120